### DISCUSSION TITLE
Skia how-to

### DISCUSSION THREAD

**Discussion by sonicsmooth** [AvaloniaUI/Avalonia#10135] [General]
Hi, is there a summary or basic example of how to interact with skia in avalonia?  I'm high-level interested to learn about:
1. creating, drawing to, manipulating bitmaps and memory off-screen
2. layers
3. how the various transforms affect everything
4. What's going on inside `Render` with its drawing context
etc.

I'm under the impression that I can get access to a skia surface from the `Render` drawing context, and then presumably follow skia docs from there.  However navigating the `Avalonia.Media.DrawingContext` from inside `Render` using debugger I don't see anything that leads to a skia object.  If I create a `Avalonia.Skia.DrawingContext` using `CreateDrawingContext` separately, however, I do see some skia stuff via the debugger exploration: canvas, surface, etc.

**Comment by ahopper** [comment]
Have you seen this https://github.com/AvaloniaUI/Avalonia/blob/master/samples/RenderDemo/Pages/CustomSkiaPage.cs
